tests/monitoring/vm_monitoring.go

```diff
@@ -229,4 +234,85 @@ var _ = Describe("[Serial][sig-monitoring]VM Monitoring", Serial, decorators.Sig
+			scales = NewScaling(virtClient, []string{virtOperator.deploymentName})
+			scales.UpdateScale(virtOperator.deploymentName, int32(0))
+
+			reduceAlertPendingTime(virtClient)
+		})
+
+		AfterEach(func() {
+			scales.RestoreAllScales()
+			alerts := []string{"KubevirtVmHighMemoryUsage", "OrphanedVirtualMachineInstances", "VMCannotBeEvicted"}
+			waitUntilAlertDoesNotExist(virtClient, alerts...)
+		})
+
+		It("should fire KubevirtVmHighMemoryUsage alert", func() {
+			By("starting VMI")
+			vmi := libvmi.NewAlpine()
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(context.Background(), vmi)
+			Expect(err).ToNot(HaveOccurred(), "should start vmi")
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("expecting the VirtualMachineInstance console")
+			Expect(console.LoginToAlpine(vmi)).To(Succeed())
+
+			By("fill up the vmi pod memory")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
+			vmiPodRequestMemory := vmiPod.Spec.Containers[0].Resources.Requests.Memory().Value()
+			_, err := exec.ExecuteCommandOnPod(
+				virtClient,
+				vmiPod,
+				vmiPod.Spec.Containers[0].Name,
+				[]string{"/usr/bin/bash", "-c", fmt.Sprintf("cat <( </dev/zero head -c %d) <(sleep 150) | tail", vmiPodRequestMemory)},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for KubevirtVmHighMemoryUsage alert")
+			verifyAlertExist(virtClient, "KubevirtVmHighMemoryUsage")
+		})
+
+		It("should fire OrphanedVirtualMachineInstances alert", func() {
+			By("starting VMI")
+			vmi := tests.NewRandomFedoraVMIWithGuestAgent()
+			tests.RunVMIAndExpectLaunch(vmi, 240)
+
+			By("delete virt-handler daemonset")
+			_, err := virtClient.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred(), "should get virt-handler daemonset")
+			err = virtClient.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Delete(context.Background(), virtHandler.deploymentName, metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for OrphanedVirtualMachineInstances alert")
+			verifyAlertExist(virtClient, "OrphanedVirtualMachineInstances")
+		})
+
+		It("should fire VMCannotBeEvicted alert", func() {
+			By("starting non-migratable VMI with eviction strategy set to LiveMigrate ")
+			vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{
+				{
+					Name: "default",
+					InterfaceBindingMethod: v1.InterfaceBindingMethod{
+						Bridge: &v1.InterfaceBridge{},
+					},
+				},
+			}
+			strategy := v1.EvictionStrategyLiveMigrate
+			vmi.Spec.EvictionStrategy = &strategy
+
+			vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 240)
+
+			By("waiting for VMCannotBeEvicted alert")
+			verifyAlertExist(virtClient, "VMCannotBeEvicted")
+
+			By("deleting the VirtualMachineInstance")
```

Comment:
Do you need to delete the VMI for the alert to disappear or could we leave that to the usual cleanup done after tests?

Reply:
removed